tech-specs/src/explore/design/pos.md

```diff
@@ -0,0 +1,268 @@
+- *self-bond*:
+  A validator may lock-up tokens into a [bond](#bonds) only for its own validator's address.
+- *unbond*:
+  Any self-bonded tokens may be partially or fully [unbonded](#unbond).
+- *withdraw unbonds*:
+  Unbonded tokens may be withdrawn in or after the [unbond's epoch](#unbond).
+- *change consensus key*:
+  Set the new consensus key. When applied in epoch `n`, the key is set for epoch `n + pipeline_length`.
+
+#### Active validator set
+
+From all the *candidate* validators, in each epoch the ones with the most voting power limited up to the `max_active_validators` [parameter](#system-parameters) are selected for the active validator set. The active validator set selected in epoch `n` is set for epoch `n + pipeline_length`.
+
+### Delegator
+
+A delegator may have any number number of delegations. Delegations are stored in [bonds](#bonds).
+
+#### Delegator actions
+
+- *delegate*:
+  An account which is not a validator may delegate tokens to any number of validators. This will lock-up tokens into a [bond](#bonds).
+- *undelegate*:
+  Any delegated tokens may be partially or fully [unbonded](#unbond).
+- *withdraw unbonds*:
+  Unbonded tokens may be withdrawn in or after the [unbond's epoch](#unbond).
+
+## Bonds
+
+A bond locks-up tokens from validators' self-bonding and delegators' delegations. For self-bonding, the source address is equal to the validator's address. Only validators can self-bond. For a bond created from a delegation, the bond's source is the delegator's account.
+
+For each epoch, bonds are uniquely identified by the pair of source and validator's addresses. A bond created in epoch `n` is written into epoch `n + pipeline_length`. If there already is a bond in the epoch `n + pipeline_length` for this pair of source and validator's addresses, its tokens are incremented by the newly bonded amount.
+
+Any bonds created in epoch `n` increment the bond's validator's total bonded tokens by the bond's token amount and update the voting power for epoch `n + pipeline_length`.
+
+The tokens put into a bond are immediately deducted from the source account.
+
+### Unbond
+
+An unbonding action (validator *unbond* or delegator *undelegate*) requested by the bond's source account in epoch `n` creates an "unbond" with epoch set to `n + unbounding_length`. We also store the epoch of the bond(s) from which the unbond is created in order to determine if the unbond should be slashed if a fault occurred within the range of bond epoch (inclusive) and unbond epoch (exclusive).
+
+Any unbonds created in epoch `n` decrements the bond's validator's total bonded tokens by the bond's token amount and update the voting power for epoch `n + unbonding_length`.
+
+An "unbond" with epoch set to `n` may be withdrawn by the bond's source address in or any time after the epoch `n`. Once withdrawn, the unbond is deleted and the tokens are credited to the source account.
+
+### Staking rewards
+
+TODO
```

Comment:
Shall we discuss this? Staking rewards are minted each block and paid to validator accounts. For now, we can use a very simple reward formula.

Reply:
I've added a simple description based on a PoS system parameters